components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2123,33 +2123,51 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         }
     }
 
-    protected defaultSpendingLimit = 100;
-    async subscribeTeamToStripe(ctx: TraceContext, teamId: string, setupIntentId: string): Promise<void> {
-        this.checkAndBlockUser("subscribeUserToStripe");
-        const team = await this.guardTeamOperation(teamId, "update");
-        await this.ensureStripeApiIsAllowed({ team });
+    async subscribeToStripe(ctx: TraceContext, attributionId: string, setupIntentId: string): Promise<void> {
+        const attrId = AttributionId.parse(attributionId);
+        if (attrId === undefined) {
+            log.error(`Invalid attribution id: ${attributionId}`);
+            throw new ResponseError(ErrorCodes.BAD_REQUEST, `Invalid attibution id: ${attributionId}`);
+        }
+
+        const user = this.checkAndBlockUser("subscribeToStripe");
+
+        let customer: Stripe.Customer | undefined;
         try {
-            let customer = await this.stripeService.findCustomerByTeamId(team!.id);
+            if (attrId.kind === "team") {
+                const team = await this.guardTeamOperation(attrId.teamId, "update");
+                await this.ensureStripeApiIsAllowed({ team });
+                customer = await this.stripeService.findCustomerByTeamId(team!.id);
+            } else {
+                await this.ensureStripeApiIsAllowed({ user });
+                customer = await this.stripeService.findCustomerByUserId(user.id);
+            }
             if (!customer) {
-                throw new Error(`No Stripe customer profile for team '${team.id}'`);
+                throw new Error(`No Stripe customer profile for '${attributionId}'`);
             }
+
             await this.stripeService.setDefaultPaymentMethodForCustomer(customer, setupIntentId);
             await this.stripeService.createSubscriptionForCustomer(customer);
-
-            const attributionId: AttributionId = { kind: "team", teamId };
-
-            // Creating a cost center for this team
             await this.usageServiceClientProvider.getDefault().setCostCenter({
```

Comment:
Out of curiosity, how would we handle the case where we do setup the user in stripe but fail to set the cost center due to downstream issues?

Reply:
Currently, I don't think we do handle it. I think we'd end up with a valid Stripe customer for which we aren't able to look up cost center information.